roles/tftp-server/tasks/pxe.yml

```diff
@@ -0,0 +1,103 @@
+---
+- name: install the tftp server
+  apt:
+    name: dnsmasq
+
+- name: enable tftp
+  copy:
+    src: files/tftp.conf
+    dest: /etc/dnsmasq.d
+
+- name: create boot image directories
+  file:
+    path: /srv/tftp/{{item}}
+    state: directory
+    recurse: true
+    mode: 0755
+  with_items:
+    - ipxe
+    - util/cz/exper/live
+
+- name: magicly build /srv/tftp/boot/grub
+  command: grub-mknetdir
+
+- name: write grub.cfg
+  template:
+    src: grub.cfg.j2
+    dest: /srv/tftp/boot/grub/grub.cfg
+
+- name: create debian suite dirs
+  file:
+    path: /srv/tftp/debian/{{ item[0] }}/{{ item[1] }}
+    state: directory
+    recurse: true
+    mode: 0755
+  with_nested:
+    - "{{ debian_suites }}"
+    - "{{ archs }}"
+
+
+- name: create utunbu suite dirs
+  file:
+      # path: /srv/tftp/ubuntu/{{suite}}/{{arch}}
+    path: /srv/tftp/ubuntu/{{ item[0] }}/{{ item[1] }}
+    state: directory
+    recurse: true
+    mode: 0755
+  with_nested:
+    - "{{ ubuntu_suites }}"
+    - "{{ archs }}"
+
+- name: download debian netboot images
+  get_url:
+    url: http://ftp.debian.org/debian/dists/{{item[0]}}/main/installer-{{item[1]}}/current/images/netboot/debian-installer/{{item[1]}}/{{item[2]}}
```

Comment:
Didn't we have a variable for the hostname?

Reply:
yep, fixed